### PR TITLE
Add Hyperspace version annotations to `Microsoft.Spark.Extensions.Hyperspace` APIs.

### PR DESCRIPTION
**Pull request by AFFogarty**
This PR adds Hyperspace version annotations to all Hyperspace APIs in `Microsoft.Spark.Extensions.Hyperspace`. 

This PR contributes to #618 .

### CODE REVIEW

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/Hyperspace.cs [hunk 1]
```diff
@@ -20,6 +21,7 @@ public class Hyperspace : IJvmObjectReferenceProvider
         private readonly IJvmBridge _jvmBridge;
         private readonly JvmObjectReference _jvmObject;
 
+        [HyperspaceSince(HyperspaceVersions.V0_0_1)]
```
Comment (Niharikadutta) [inline src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/Hyperspace.cs:24]: nit: One naive question, if the version attribute has been added on the class level, does it need to be mentioned for every class API?

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/HyperspaceVersions.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.Spark.Extensions.Hyperspace
+{
+    internal static class HyperspaceVersions
+    {
+        internal const string V0_0_1 = "0.0.1";
+        internal const string V0_0_2 = "0.0.2";
```
Comment (Niharikadutta) [inline src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/HyperspaceVersions.cs:10]: I don't see this version being attributed anywhere, is there a reason we have included it here?

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/Hyperspace.cs (hunk 1 above)
Comment (AFFogarty) [reply src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/Hyperspace.cs:24]: Yeah. For example, a later version of Hyperspace could add some new functions to an existing class.

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/HyperspaceVersions.cs (hunk 2 above)
Comment (AFFogarty) [reply src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/HyperspaceVersions.cs:10]: I included it just for completeness.  @suhsteve Do you think we should include only the necessary versions?

Comment (suhsteve) [reply src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/HyperspaceVersions.cs:10]: We can just add new versions as necessary, ie when we need to use it.

Comment (AFFogarty) [reply src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/HyperspaceVersions.cs]: Sounds good, removed `0.0.2`.

**Review by suhsteve — APPROVED** [review]
LGTM.  Thanks @AFFogarty

**Review by imback82 — APPROVED** [review]
LGTM, thanks @AFFogarty!

**Pull request merged**